Client/Frontend/Settings/HomepageSettings/WallpaperSettings/v1/WallpaperBaseViewController.swift

```diff
@@ -0,0 +1,53 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+
+class WallpaperBaseViewController: UIViewController {
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass
+            || previousTraitCollection?.verticalSizeClass != traitCollection.verticalSizeClass {
+            updateOnRotation()
+        }
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        updateOnRotation()
+    }
+
+    /// On iPhone, we call updateOnRotation when the trait collection has changed, to ensure calculation
+    /// is done with the new trait. On iPad, trait collection doesn't change from portrait to landscape (and vice-versa)
+    /// since it's `.regular` on both. We updateOnRotation from viewWillTransition in that case.
+    func updateOnRotation() {
+    }
```

Comment:
Shouldn't this be better served as an optional protocol function rather than having it part of the base class where it isn't implemented?

Reply:
As far as I know that wouldn't if I want to call the method in this class (from traitCollectionDidChange and viewWillTransition).